rmgpy/data/thermo.py

```diff
@@ -1251,6 +1261,170 @@ def getThermoData(self, species, trainingSet=None):
+        self.deltaAtomicAdsorptionEnergy = deltaAtomicAdosrptionEnergy
+
+    def correctBindingEnergy(self, thermo, species):
+        """
+        Changes the provided thermo, by applying a linear scaling relation
+        to correct the adsorption energy.
+
+        :param thermo: starting thermo data
+        :param species: the species (which is an adsorbate)
+        :return: corrected thermo
+        """
+        molecule = species.molecule[0]
+        # only want/need to do one resonance structure
+        surfaceSites = []
+        for atom in molecule.atoms:
+            if atom.isSurfaceSite():
+                surfaceSites.append(atom)
+        normalizedBonds = {'C':0., 'O':0., 'N':0., 'H':0.}
+        maxBondOrder = {'C':4., 'O':2., 'N':3., 'H':1.}
+        for site in surfaceSites:
+            numbonds = len(site.bonds)
+            if numbonds == 0:
+                #vanDerWaals
+                pass
+            else:
+                assert len(site.bonds) == 1, "Each surface site can only be bonded to 1 atom"
+                bondedAtom = site.bonds.keys()[0]
+                bond = site.bonds[bondedAtom]
+                if bond.isSingle():
+                    bondOrder = 1.
+                elif bond.isDouble():
+                    bondOrder = 2.
+                elif bond.isTriple():
+                    bondOrder = 3.
+                elif bond.isQuadruple():
+                    bondOrder = 4.
+                else:
+                    raise NotImplementedError("Can't remove surface bond of type {}".format(bond.order))
+
+                normalizedBonds[bondedAtom.symbol] += bondOrder / maxBondOrder[bondedAtom.symbol]
+
+        if not isinstance(thermo, ThermoData):
+            thermo = thermo.toThermoData()
+            findCp0andCpInf(species, thermo)
+
+        ## now edit the adsorptionThermo using LSR
+        for element in 'CHON':
+            changeInBindingEnergy = self.deltaAtomicAdsorptionEnergy[element].value_si * normalizedBonds[element]
+            thermo.H298.value_si += changeInBindingEnergy
+        thermo.comment += " Binding energy corrected by LSR."
+        return thermo
+
+
+
+    def getThermoDataForSurfaceSpecies(self, species):
+        """
+        Get the thermo data for an adsorbed species,
+        by desorbing it, finding the thermo of the gas-phase
+        species, then adding an adsorption correction that
+        is found from the groups/adsorption tree.
+        Does not apply linear scaling relationship.
+        
+        Returns a :class:`ThermoData` object, with no Cp0 or CpInf
+        """
+
+        assert not species.isSurfaceSite(), "Can't estimate thermo of vacant site. Should be in library (and should be 0)"
+
+        logging.debug(("Trying to generate thermo for surface species"
+                        " with these {} resonance isomer(s):").format(len(species.molecule)))
```

Comment:
This doesn't really seem like a useful debug message. Was the intention to log the adjlists as well?